frame/democracy/src/lib.rs

```diff
@@ -1866,31 +1843,34 @@ impl<T: Config> Pallet<T> {
 		let preimage = <Preimages<T>>::take(&proposal_id);
 		if let Some(PreimageStatus::Available { data, provider, deposit, .. }) = preimage {
 			if let Ok(proposal) = T::Proposal::decode(&mut &data[..]) {
-				let err_amount = T::NativeCurrency::release(&provider, deposit, true)?;
-				debug_assert_eq!(err_amount, deposit);
-				Self::deposit_event(Event::<T>::PreimageUsed(proposal_id.hash, provider, deposit));
-				let signed_raw_origin = T::GovernanceRegistry::get(&proposal_id.asset_id)?;
-				let raw_origin: frame_system::RawOrigin<T::AccountId> = signed_raw_origin.into();
-				let res = proposal.dispatch(raw_origin.into()).map(|_| ()).map_err(|e| e.error);
-				Self::deposit_event(Event::<T>::Executed(index, res));
+				let err_amount = T::NativeCurrency::unreserve(&provider, deposit);
+				debug_assert!(err_amount.is_zero());
+				Self::deposit_event(Event::<T>::PreimageUsed {
+					proposal_hash: proposal_id.hash,
+					provider,
+					deposit,
+				});
+
+				let res = proposal
+					.dispatch(frame_system::RawOrigin::Root.into())
```

Comment:
We need to use the governance registry for this, we cannot simply dispatch using Root for any AssetId. Please have a look at how the previous fork of the democracy pallet handles this. Alternatively, we can get rid of the governance registry, and instead convert the AssetId to an AccountId, and use that to dispatch.

Reply:
that would allow us to deprecate the governance-registry

Reply:
Okay , I'm lookt at that and make necessary updates to this

Reply:
This has been resolved